app/controllers/responses_controller.rb

```diff
@@ -0,0 +1,155 @@
+      end
+    when 'unsubmit', 'destroy'
+      unless has_role?('Instructor') || has_role?('Admin')
+        render json: { error: 'forbidden' }, status: :forbidden
+      end
+    else
+      render json: { error: 'forbidden' }, status: :forbidden
+    end
+    true
+  end
+
+  # POST /responses
+  def create
+    @response_map = ResponseMap.find_by(id: params[:response_map_id] || params[:map_id])
+    return render json: { error: 'ResponseMap not found' }, status: :not_found unless @response_map
+
+    @response = Response.new(
+      map_id: @response_map.id,
+      is_submitted: false,
+      created_at: Time.current
+    )
+
+    if @response.save
+      render json: { message: 'Response draft created successfully', response: @response }, status: :created
+    else
+      render json: { error: @response.errors.full_messages.to_sentence }, status: :unprocessable_entity
+    end
+  end
+
+  # PATCH /responses/:id
+  # Reviewer edits existing draft (still unsubmitted)
+  def update
+    puts "Updating Response ID: #{params[:id]}"
+    return render json: { error: 'forbidden' }, status: :forbidden if @response.is_submitted?
+
+    if @response.update(response_params)
+      render json: { message: 'Draft updated successfully', response: @response }, status: :ok
+    else
+      render json: { error: @response.errors.full_messages.to_sentence }, status: :unprocessable_entity
+    end
+  end
+
+  # PATCH /responses/:id/submit
+  # Lock the response and calculate final score
+  def submit
+    return render json: { error: 'Response not found' }, status: :not_found unless @response
+    return render json: { error: 'Response already submitted' }, status: :unprocessable_entity if @response.is_submitted?
+
+    # Check deadline
+    return render json: { error: 'Deadline has passed' }, status: :forbidden unless deadline_open?(@response)
+
+    # Validate rubric completion
+    unanswered = @response.scores.select { |a| a.answer.nil? }
+    return render json: { error: 'All rubric items must be answered' }, status: :unprocessable_entity unless unanswered.empty?
+
+    # Lock response
+    @response.is_submitted = true
+
+    # Calculate score via ScorableHelper
+    total_score = @response.aggregate_questionnaire_score
+
+    if @response.save
+      render json: {
+        message: 'Response submitted successfully',
+        response: @response,
+        total_score: total_score
+      }, status: :ok
+    else
+      render json: { error: @response.errors.full_messages.to_sentence }, status: :unprocessable_entity
+    end
+  end
+
+  # PATCH /responses/:id/unsubmit
+  # Instructor/Admin reopens a submitted response
+  def unsubmit
+    return render json: { error: 'Response not found' }, status: :not_found unless @response
+
+    if @response.is_submitted?
+      @response.update(is_submitted: false)
+      render json: { message: 'Response reopened for revision', response: @response }, status: :ok
```

Comment:
The word "Response" here could be confusing.  That's what it is in the system, but the UI should use better names.  Don't we have print names for each different kind of Questionnaire?  Also, "reopened for revision" is much too cryptic.  Say it in plain English!